### DISCUSSION TITLE
Text with inline elements

### DISCUSSION THREAD

**Discussion by ickby** [zauberzeug/nicegui#3236] [Q&A] [answered]
### Question

NiceGUI and Vue noob here. HTML provides the nice feature to inline any HTML element into text. Is there any way to do this with nicegui, to add arbitrary complex elements into a text line? Some like this pseudo code?

```py
with ui.textline():'
    ui.text("This is my sentence")
    ui.button(" with an inline button")
    ui.text("that aligns and sizes nicely")
```

I know that I can add a ui.html element with inlined html tags, but I would like to use nicegui elements for inlining, as they can become rather complex. If not possible by default, any idea how I could create such an element?

**Comment by rodja** [comment, marked as answer]
Have a look at https://nicegui.io/documentation/teleport.

**Comment by ickby** [comment]
nice, that worked well:

```py
base = ui.html("This is a <a></a> in my text")
with ui.teleport(f'#c{base.id} a'):
    ui.button("button")
```

thanks for the hint.

**Comment by BlankAdventure** [comment]
I know this has been closed but couldn't the OPs request be largely accomplished with this?

```python
with ui.row().classes('items-center'):
    ui.label("This is my sentence")
    ui.button(" with an inline button")
    ui.label("that aligns and sizes nicely")
```

**Reply by falkoschindler** [reply]
That's what I thought as well. But I would use `ui.element` instead of `ui.row` to avoid using a flex layout. And I guess the children need to get an "inline" class for their text to wrap, even though the button won't wrap anyway:

```py
with ui.element().classes('w-[400px] bg-gray-100'):
    ui.label('This is my sentence ').classes('inline')
    ui.button('with an inline button').classes('inline').props('flat dense')
    ui.label(' that aligns and sizes nicely.').classes('inline')
```

<img width="416" alt="Screenshot 2024-06-21 at 15 55 46" src="https://github.com/zauberzeug/nicegui/assets/5767091/5e7d326f-2413-4e43-b40b-e23940254368">

**Reply by ickby** [reply]
nice! ui.row() I tried before, but that wraps not very natural, so I don't like that variant. But the classes('inline') seem to work very well, thats a option I did not figure out.  Thanks!
